packages/runtime/garbage-collector/src/utils.ts

```diff
@@ -4,133 +4,6 @@
-	const clonedGCNodes: { [id: string]: string[] } = {};
-	for (const [id, outboundRoutes] of Object.entries(gcData.gcNodes)) {
-		clonedGCNodes[id] = Array.from(outboundRoutes);
-	}
-	return {
-		gcNodes: clonedGCNodes,
-	};
-}
-
-/**
- * Helper function that unpacks the GC details of the children from a given node's GC details.
- * @param gcDetails - The GC details of a node.
- * @returns A map of GC details of each children of the the given node.
- *
- * @deprecated Internal implementation detail and will no longer be exported in an upcoming release.
- */
-export function unpackChildNodesGCDetails(gcDetails: IGarbageCollectionDetailsBase) {
-	const childGCDetailsMap: Map<string, IGarbageCollectionDetailsBase> = new Map();
-
-	// If GC data is not available, bail out.
-	if (gcDetails.gcData === undefined) {
-		return childGCDetailsMap;
-	}
-
-	const gcNodes = gcDetails.gcData.gcNodes;
-	for (const [id, outboundRoutes] of Object.entries(gcNodes)) {
-		// Skip self-node since only children GC data is to be generated.
-		if (id === "/") {
-			continue;
-		}
-
-		assert(id.startsWith("/"), 0x2ae /* "node id should always be an absolute route" */);
-		const childId = id.split("/")[1];
-		let childGCNodeId = id.slice(childId.length + 1);
-		// GC node id always begins with "/". Handle the special case where a child's id in the parent's GC nodes is
-		// of format `/root`. In this case, the childId is root and childGCNodeId is "". Make childGCNodeId = "/".
-		if (childGCNodeId === "") {
-			childGCNodeId = "/";
-		}
-
-		let childGCDetails = childGCDetailsMap.get(childId);
-		if (childGCDetails === undefined) {
-			childGCDetails = { gcData: { gcNodes: {} }, usedRoutes: [] };
-		}
-		// gcData should not undefined as its always at least initialized as  empty above.
-		assert(
-			childGCDetails.gcData !== undefined,
-			0x2af /* "Child GC data should have been initialized" */,
-		);
-		childGCDetails.gcData.gcNodes[childGCNodeId] = [...new Set(outboundRoutes)];
-		childGCDetailsMap.set(childId, childGCDetails);
-	}
-
-	if (gcDetails.usedRoutes === undefined) {
-		return childGCDetailsMap;
-	}
-
-	// Remove the node's self used route, if any, and generate the children used routes.
-	const usedRoutes = gcDetails.usedRoutes.filter((route) => route !== "" && route !== "/");
-	for (const route of usedRoutes) {
-		assert(route.startsWith("/"), 0x2b0 /* "Used route should always be an absolute route" */);
-		const childId = route.split("/")[1];
-		const childUsedRoute = route.slice(childId.length + 1);
-
-		const childGCDetails = childGCDetailsMap.get(childId);
-		assert(
-			childGCDetails?.usedRoutes !== undefined,
-			0x2b1 /* "This should have be initiallized when generate GC nodes above" */,
-		);
-
-		childGCDetails.usedRoutes.push(childUsedRoute);
-		childGCDetailsMap.set(childId, childGCDetails);
-	}
-	return childGCDetailsMap;
-}
 
 /**
  * Helper function that unpacks the used routes of children from a given node's used routes.
```

Comment:
If `unpackChildNodesUsedRoutes` is the only remaining member of this file, maybe rename it to `unpackChildNodesUsedRoutes.ts` to deter new additions (I feel like things called "utils" accumulate things they often should not).

Reply:
I am also wondering if I should move it to an existing runtime-util file and remove this package altogether. What's your though on that one?

Reply:
I'd support that.  I think longer term would be good to collapse datastore into container-runtime and then it could go into that merged package?  But for now consolidating into runtime-utils sounds good.

Reply:
I don't think we can collapse datastore into container-runtime because they are different layers,

Reply:
I am going to remove garbage-collector package then. @tylerbutler Please hold on creating a changeset for this just yet.

Reply:
> I don't think we can collapse datastore into container-runtime because they are different layers,

Ultimately they're both in the runtime layer.  I know some effort has gone into trying to establish layers here but just looking at it I don't think we've really achieved a separable layer (e.g. imagine trying to implement an alternative `FluidDataStoreRuntime` with any success).  Not something that needs to be resolved immediately though.

Reply:
What I meant is data store runtime is a separate compatibility layer and we support dynamic loading of data stores. Our tests validate N / N-1 support for them. So, not sure how we can merge them into one layer.